core/types/transaction.go

```diff
@@ -286,9 +265,23 @@ func (tx *Transaction) DeserializeUnsigned(r io.Reader) error {
 	if err != nil {
 		return err
 	}
+	gasPrice, err := serialization.ReadUint64(r)
+	if err != nil {
+		return err
+	}
+	gasLimit, err := serialization.ReadUint64(r)
+	if err != nil {
+		return err
+	}
+	payer := new(common.Address)
+	payer.Deserialize(r)
```

Comment:
handle error, and you can use `tx.Payer.Deserialize` directly to avoid allocation.